cirq-google/cirq_google/workflow/quantum_runtime.py

```diff
@@ -174,15 +255,29 @@ def execute(
         # coverage: ignore
         raise ValueError("Please provide a non-empty `base_data_dir`.")
 
-    os.makedirs(f'{base_data_dir}/{run_id}', exist_ok=False)
-
-    # Results object that we will fill in in the main loop.
-    exegroup_result = ExecutableGroupResult(
-        runtime_configuration=rt_config,
-        shared_runtime_info=SharedRuntimeInfo(run_id=run_id),
-        executable_results=list(),
+    # Set up data saving, save runtime configuration.
+    data_dir = f'{base_data_dir}/{run_id}'
+    os.makedirs(data_dir, exist_ok=False)
+    egr_record = ExecutableGroupResultFilesystemRecord(
+        runtime_configuration_fn='QuantumRuntimeConfiguration.json.gz',
+        shared_runtime_info_fn='SharedRuntimeInfo.json.gz',
+        executable_result_fns=[],
+        run_id=run_id,
     )
-    cirq.to_json_gzip(exegroup_result, f'{base_data_dir}/{run_id}/ExecutableGroupResult.json.gz')
+    cirq.to_json_gzip(rt_config, f'{data_dir}/{egr_record.runtime_configuration_fn}')
+
+    # Set up to-be-updated objects.
+    shared_rt_info = SharedRuntimeInfo(run_id=run_id)
+    _update_updatable_files(egr_record, shared_rt_info, data_dir)
+
+    executable_results = []
+
+    def _finalize_exe_result(exe_result: ExecutableResult, i: int):
+        """Do all the bookkeeping when an ExecutableResult has been completed."""
+        exe_result_fn = f'ExecutableResult.{i}.json.gz'
+        cirq.to_json_gzip(exe_result, f"{data_dir}/{exe_result_fn}")
+        executable_results.append(exe_result)
+        egr_record.executable_result_fns.append(exe_result_fn)
```

Comment:
Nit: Why is this a seperate nested function if it's only getting called into once ?

Reply:
eh I thought giving it a name would make the main loop more readable but probably not.

Reply:
done. unnested